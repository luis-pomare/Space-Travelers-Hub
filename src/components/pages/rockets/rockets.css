.rocket-section {
  display: flex;
  margin: 60px 60px 60px 60px;
}

.rocket-description {
  display: flex;
  flex-direction: column;
}

h2 {
  font-size: 20px;
}

.images {
  width: 40%;
  height: 40%;
}

.reserve-button {
  width: 150px;
}
