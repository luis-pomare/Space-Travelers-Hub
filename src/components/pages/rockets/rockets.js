import { useSelector } from 'react-redux';
import NewRocket from '../../newRocket/newRocket';

export default function Rockets() {
  const { rockets } = useSelector((state) => state.rockets);
  const rocketList = Object.keys(rockets);

  return (
    <section className="list-of-rockets">
      {
        rocketList ? rocketList.map((rocket) => (
          <NewRocket
            key={parseFloat(rocket)}
            id={parseFloat(rocket)}
            rocketName={rockets[rocket].rocketName}
            description={rockets[rocket].description}
            flickrImages={rockets[rocket].flickrImages}
            reserved={rockets[rocket].reserved}
          />
        )) : <p>Rockets not found</p>
      }
    </section>
  );
}
