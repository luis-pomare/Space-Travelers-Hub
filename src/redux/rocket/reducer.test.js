import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../configureStore';
import rockets from './rocket';
import Rockets from '../../components/pages/rockets/rockets';
import getRockets from '../../rocket-api';

const renderReservedRockets = (reserved) => {
  const { tree } = render(
    <Provider store={store}>
      <Rockets
        reserved={reserved}
        id={0}
        rocketName="Falcon 1"
        description="Test description"
        flickrImages="https://imgur.com/DaCfMsj.jpg"
      />
    </Provider>,
  );
  return tree;
};

test('Renders first Rocket', () => {
  expect(renderReservedRockets(true)).toMatchSnapshot();
});

test('should return the initial state', () => {
  expect(rockets.reducer(undefined, { type: undefined })).toEqual(
    { rockets: [], status: null },
  );
});

test('should return the loading state', () => {
  const loadingState = { rockets: [], status: 'loading' };

  expect(rockets.reducer(loadingState, getRockets('loading'))).toEqual(
    { rockets: [], status: 'loading' },
  );
});

test('should return the success state', () => {
  const loadingState = { rockets: [], status: 'success' };

  expect(rockets.reducer(loadingState, getRockets('success'))).toEqual(
    { rockets: [], status: 'success' },
  );
});

test('should return the failed state', () => {
  const loadingState = { rockets: [], status: 'failed' };

  expect(rockets.reducer(loadingState, getRockets('failed'))).toEqual(
    { rockets: [], status: 'failed' },
  );
});
