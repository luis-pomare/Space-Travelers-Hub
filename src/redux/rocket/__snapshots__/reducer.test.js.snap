// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders first Rocket 1`] = `undefined`;
